docs/kubebuilder_v0_v1_difference.md

```diff
@@ -0,0 +1,124 @@
+# Kubebuilder v0 v.s. v1
```

Comment:
We should have a line explaining what `v0` and `v1` really is. Explanation should highlight that `v1` project layout is architecturally different from `v1` layout. `v1` project uses `controller-runtime` set of libraries for controller implementation and used tools under `controller-tools` for scaffold and generation. It will help the users with the context.

Reply:
Added a paragraph to explain the meaning of v0 and v1 projects.